### PR TITLE
Migrating MassEditing to v10

### PR DESCRIPTION
**Pull request by JayVora-SerpentCS**

### CODE REVIEW

**Comment by nhomar** [comment]
Thanks @JayVora-SerpentCS 

One quick quiestion.

I think the button in the screen that create the action to comply with the >v8.0 usability standard due to the fact that is an "action" button and not a "link" buttton should be in the top of the screen.

I know this is a silly question but I personally think we all will be more happy if we migrate to 10.0 with this little change..



Can You? or do you prefer I make a separate PR?

**Review by jgrandguillaume — APPROVED** [review]
Hi @JayVora-SerpentCS 

Thanks for this great module ! 

Tested and works well :+1: 

Regards,

Joël

**Comment by atchuthan** [comment]
Found a strange bug when I tried to test the functionality in Runbot

URL: http://3271979-598-6ab5a8.runbot1.odoo-community.org/

Partner Mass editing
image

Mass editing window after selecting 2 partners (ASUSTeK/Test1 & ASUSTeK/Test2)
image

Mass editing window after selecting all partners
image

**Comment by pedrobaeza** [comment]
@JayVora-SerpentCS, thanks for maintaining this module. Please follow full 10.0 migration guide: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0#tasks-to-do-in-the-migration and take care about the comments.

You also need to bring the latest changes of the 9.0 branch. For example https://github.com/OCA/server-tools/commit/65a1579e99ff623bf39e3e081baa9d31f0a62736. This is done with https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0#technical-method-to-migrate-a-module-from-90-to-100-branch

**Comment by JayVora-SerpentCS** [comment]
@pedrobaeza how to bring https://github.com/OCA/server-tools/commit/65a1579e99ff623bf39e3e081baa9d31f0a62736?

**Comment by pedrobaeza** [comment]
Do:
```
git remote update
git cherry-pick 65a1579
```

**Review by njeudy — APPROVED** [review]
Tested and code review

👍

**Review by antoniocanovas — APPROVED** [review]
LGTM, Tested on runbot.

**Comment by pedrobaeza** [comment]
The missing commits are still not brought and there's a comment unattended.

**Comment by JayVora-SerpentCS** [comment]
Agreed, working on them soon.
Thanks.

**Comment by ovnicraft** [comment]
Hi @JayVora-SerpentCS any updates for this ? is ready to merge ? @pedrobaeza

**Comment by JayVora-SerpentCS** [comment]
Cherrypicked Pedro's commit. now working on the other 2 comments.

**Comment by JayVora-SerpentCS** [comment]
@nhomar covered.
screenshot from 2016-12-29 16-13-45
@atchuthan I tried various scenarios, but did not manage to reproduce. Would you please retry?
Thanks.

**Comment by ovnicraft** [comment]
Tested on runbot and working ok 👍 
I cant reproduce @atchuthan bug reported in partners.

**Comment by ovnicraft** [comment]
There is no report to reproduce the bug mentioned by @atchuthan, IMO this is ready to merge.

**Review by JonathanNEMRY — CHANGES_REQUESTED** [review]
Hi! Great work! I just made a PR on your branch to add setuptools. Otherwise :+1: 
https://github.com/JayVora-SerpentCS/server-tools/pull/1

**Review by JonathanNEMRY — APPROVED** [review]
:+1:

**Comment by pedrobaeza** [comment]
Squashing and merging. Thanks for the work.

**Comment by pedrobaeza** [comment]
@JayVora-SerpentCS, I'm missing these two important pieces for improving the module that has been done in v8:

* https://github.com/OCA/server-tools/commit/444242489199b5a50213e2f79791d60aad00949d
* https://github.com/OCA/server-tools/commit/09732a891c96d625b31b631995f3adf3d2aff4f3

Can you please adapt the code for this version?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Due to above comment

**Comment by JayVora-SerpentCS** [comment]
Processed.

**Review by atchuthan — APPROVED** [review]
@JayVora-SerpentCS 
Sorry for the late follow-up.  I couldn't reproduce the error in latest runbot.  So, :+1: my vote

**Comment by pedrobaeza** [comment]
@JayVora-SerpentCS you have the bad habit of overwriting author commits with yours. Please respect the credits of the person who added the last 2 commits. What you have to do is cherry-picking commits as is, solving any possible conflict, and then adding a new commit with the changes you need to do for adapting to v10 (excluding conflicts already solved).

Please also see for the runbot warning that appears (I have launched a rebuild for seeing the log). When done, I'll merge.

**Comment by JayVora-SerpentCS** [comment]
@pedrobaeza , no we did not overwrite anyone's commit. We cherry picked only.
We will check what's overwritten. If not, we will propose a clean merge.

**Comment by JayVora-SerpentCS** [comment]
Help me understand better, the cherrypicked commits are already linked and referred : https://github.com/OCA/server-tools/pull/598/commits/6c7e2402a750579f07e826246efaca547bac5238
If you think cleaning is better, I will clean the proposal and push in single commit after cherrypicking again.

**Comment by pedrobaeza** [comment]
This is not cherry-picking, Jay. A cherry-pick is that you bring the commits as is, not integrating them in another commit with you as the author of the full diff:

seleccion_001

A correct cherry-pick would show in the commit list both commits with the corresponding authors, something like:

seleccion_002

**Comment by mourad-ehm** [comment]
Code review :+1:

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]
This error pops when trying to add items to a new mass.object record (testing on runbot):
image

Also, please attend runbot warning: <code>WARNING openerp_test odoo.fields: Field mass.object.name: parameter 'select' is no longer supported; use 'index' instead.</code>

**Comment by JayVora-SerpentCS** [comment]
@pedrobaeza You are a nice teacher along with Google! I learned well now the cherry-picking. Earlier commits were squashed and hence it would reflect we ate them, which we actually did not.
I rebased the PR nicely now.
@lreficent Sorry I do not see a reason to face the error. Any detailed traceback possible?
Thanks.

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
When creating a mass editing record on runbot I'm getting an error:
image

**Comment by LoisRForgeFlow** [comment]
@JayVora-SerpentCS I get the error when trying to create a Mass Editing record as usual and trying to add the fields. I have just tested on local right now, same error raises. Aren't you getting the error? Maybe it is my fault but I think it is worth that you test it at least functionally in a moment to test whether it pops for you or not. Really appreciate the effort, thanks!

**Comment by JayVora-SerpentCS** [comment]
Will check definitely soon and fix if such happens.

**Comment by JayVora-SerpentCS** [comment]
@aheficent @lreficent True that, made sure no loop hole stays now, edited the domain and fixed.
Thanks.

File: mass_editing/models/mass_object.py [hunk 1]
```diff
@@ -5,6 +5,16 @@
 from odoo import api, fields, models, _
 
 
+class Fields(models.Model):
+    _inherit = 'ir.model.fields'
+
+    @api.model
+    def search(self, args, offset=0, limit=0, order=None, count=False):
+        print args
```
Comment (AaronHForgeFlow) [inline mass_editing/models/mass_object.py]: I guess this is here for debugging and you will remove it. Also, this is causing travis to fail.

The runbot is working now. Thanks.

Comment (JayVora-SerpentCS) [reply mass_editing/models/mass_object.py]: For sure, thanks, managed. Had pushed in a hurry :)

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Thanks

**Review by LoisRForgeFlow — APPROVED** [review]
Works fine now. Thank you!

**Comment by pedrobaeza** [comment]
@JayVora-SerpentCS, if you want to avoid the forgotten print and pylint/flake8 errors locally, you can install the GIT pre-commit we prepare in MQT: https://github.com/OCA/maintainer-quality-tools/tree/master/git

**Comment by JayVora-SerpentCS** [comment]
@pedrobaeza I am using pycharm with pep8 and pyflakes extension already.
I will install the precommit too. Thanks.

**Review by pedrobaeza — APPROVED** [review]
Well, finally the history is still no clean (as the other commits don't contain diff), but I don't want to block more time, so I finally squash.

**Pull request merged**